languages/js/src/Oso.ts

```diff
@@ -2,32 +2,26 @@ import { Polar } from './Polar';
 import { Variable } from './Variable';
 import { Expression } from './Expression';
 import { Pattern } from './Pattern';
-import type {
-  Options,
-  CustomError,
-  Class,
-  PolarTerm,
-  DataFilteringQueryParams,
-} from './types';
+import type { Options, CustomError, Class, PolarTerm } from './types';
 import {
   NotFoundError,
   ForbiddenError,
   OsoError,
   UnregisteredClassError,
 } from './errors';
-import { filterData } from './dataFiltering';
-import type { FilterPlan } from './dataFiltering';
+import { parseFilter, Adapter, FilterJson } from './filter';
 
 /** The Oso authorization API. */
 // TODO(gj): maybe pass DF options to constructor & try to parametrize a
 // `Query` type w/ the return type of the provided buildQuery fn.
 export class Oso<
+  Q = unknown,
```

Comment:
`Q` and `R` seem _okay_ in `Polar` & `Host` since they're internal interfaces and there's only two generic type params, but what do you think about calling this one `Query` to match the rest of the spelled-out generic parameters here?

Reply:
6c71c2d4